### PR TITLE
Add Block Builder tests that touch chainstate

### PR DESCRIPTION
**Pull request by alfiedotwtf**

### CODE REVIEW

File: blockprod/src/detail/builder.rs [hunk 1]
```diff
@@ -214,3 +214,203 @@ impl Drop for BlockMakersDestroyer {
...
+        let mut builder = PerpetualBlockBuilder::new(
+            chain_config,
+            chainstate,
+            mempool,
+            Default::default(),
+            rx_builder,
+            true,
+        );
+
+        let block_id = Id::new(H256::from_low_u64_be(make_pseudo_rng().gen()));
```
Comment (azarovh) [inline blockprod/src/detail/builder.rs:250]: you can use `H256::random_using`

**Review by azarovh — APPROVED** [review]

File: blockprod/src/detail/builder.rs (hunk 1 above)
Comment (alfiedotwtf) [reply blockprod/src/detail/builder.rs:250]: That's a nice convenience function, thanks will update.

File: blockprod/src/detail/builder.rs [hunk 2]
```diff
@@ -214,3 +214,203 @@ impl Drop for BlockMakersDestroyer {
...
+            chainstate.call_mut(|this| {
+                this.process_block(block, BlockSource::Local).expect("Error processing block")
+            });
+        });
+
+        manager.main().await;
+    }
+
+    #[tokio::test]
+    async fn test_subscribe_to_chainstate_events() {
```
Comment (stanislav-tkach) [inline blockprod/src/detail/builder.rs:363]: Nitpick: I think the `test_` prefix can be omitted.

**Review by stanislav-tkach — APPROVED** [review]

File: blockprod/src/detail/builder.rs (hunk 2 above)
Comment (alfiedotwtf) [reply blockprod/src/detail/builder.rs]: Thanks. I've removed all the ones that shouldn't clash with real functions

**Review by TheQuantumPhysicist — APPROVED** [review]

**Review by muursh — APPROVED** [review]

**Review by iljakuklic — APPROVED** [review]

File: blockprod/src/detail/builder.rs [hunk 3]
```diff
@@ -214,3 +214,203 @@ impl Drop for BlockMakersDestroyer {
...
+
+    use super::*;
+
+    #[test]
+    fn new_tip() {
+        let (_manager, chain_config, chainstate, mempool) = setup_blockprod_test();
+
+        let (_tx_builder, rx_builder) = mpsc::unbounded_channel();
+
+        let mut builder = PerpetualBlockBuilder::new(
```
[the remark this answers is not in the document]
Comment (alfiedotwtf) [reply blockprod/src/detail/builder.rs]: I'll take any chance to avoid confusion. Done.

**Pull request merged**